src/main/kotlin/exercise3/task1/BalancedBrackets.kt

```diff
@@ -24,9 +26,21 @@ package exercise3.task1
  *
  */
 
+val brackets: Map<Char, Char> = mapOf(']' to '[', '}' to '{', ')' to '(')
 
-internal fun isExpressionBalanced(expression: String): Boolean {
-    TODO("Implement me!!!")
+fun isExpressionBalanced(expression: String): Boolean {
+    val stack = LinkedList<Char>()
+    expression.forEach {
+        when (it) {
+            in brackets.values -> stack.push(it)
```

Comment:
This is the correct implementation. 👏🏻 

One potential nitpick here is that you are creating a new list of values and keys on every item you iterate, which can be improved if we assigned `brackets.values` and `brackets.keys` to variables before we start iterating over the chars in the expression.